### PR TITLE
Fix inline record printing in outcome printer

### LINKED ISSUE #1163: Inline records in variant type definitions aren't pretty-printed properly in rtop
Reason 1.11.0 + OCaml 4.04.0 from opam

Inline record arguments to variant constructors are not pretty-printed properly from rtop but they do show up properly in actual values:

For example:

```reason
Reason # type t = | A { x : int };
type t = A
Reason # A { x : 1 };
- : t = A {x : 1}
```

`refmt` does the right thing so it seems to be limited to `rtop`.

### PR DESCRIPTION
**Pull request by anmonteiro**
This fixes #1163 and supersedes #2103

### CODE REVIEW

File: formatTest/oprintTests/expected_output/inlineRecord.re [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type t0 = T0 { t0: int, };
+type t1 = A { x: int, } | B | C { c1: string, c2: string, };
```
Comment (IwanKaramazow) [inline formatTest/oprintTests/expected_output/inlineRecord.re]: Doesn't Reason have parens around constructor args?

Comment (anmonteiro) [reply formatTest/oprintTests/expected_output/inlineRecord.re]: I don't think so. Here's an rtop session:

```rust
Reason # type a = | A {x:int}: a;
type a = A(): a;
Reason # type a = | A {x:int};
type a = A();
Reason # type a = | A({x:int});;
Error: Syntax error
Reason # type a = | A {x:int}: a;
type a = A(): a;
Reason # type a = | A ({x:int}): a;
Error: Syntax error
```

Comment (anmonteiro) [reply formatTest/oprintTests/expected_output/inlineRecord.re]: Maybe it should, though?

Comment (hcarty) [reply formatTest/oprintTests/expected_output/inlineRecord.re]: The lack of support for wrapping parens for inline records, when they're required in other cases, is very confusing.  It's an inconsistency which would be good to fix.

**Comment by anmonteiro** [comment]
@jordwalke rebased on master.

File: formatTest/oprintTests/expected_output/inlineRecord.re [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+type t2(_) =
+    D { x: int, }: t2(int)
```
Comment (jordwalke) [inline formatTest/oprintTests/expected_output/inlineRecord.re]: Great! You have a GADT test case.

File: formatTest/oprintTests/expected_output/inlineRecord.re (hunk 1 above)
Comment (jordwalke) [reply formatTest/oprintTests/expected_output/inlineRecord.re]: Yeah, I just saw someone get pretty confused about this too.
One somewhat related question I have is if there might ever be a change in OCaml that requires syntactically distinguishing values of inline variant records vs. values of variants that happen to contain records as their payload. I doubt it. That tells me that we should probably move to consistent parens everywhere.
If we hypothetically one day figured out a way to be able to drop parens for single argument things, then we would apply it consistently everywhere, not just on variants. Good discussion, thanks.

**Pull request merged**